### ISSUE TITLE
season 28 not working

### ISSUE THREAD

**Issue by Us3r-3C29x** [danielwerg/r6api.js#95]
I tried to use it, but its not working anymore

Please fix

**Comment by danielwerg** [comment]
Same.

Please open pr if you have a fix :)

I will try to figure out how to get stuff from ranked 2.0 once I have enough free time, spending most of my coding time on Advent of Code these days.

related: https://github.com/Seems2Legit/Rainbow-Six-Siege-Player-Stats-API/issues/91

**Comment by Us3r-3C29x** [comment]
Any updates?

**Comment by Us3r-3C29x** [comment]
https://github.com/hmes98318/r6s-stats-api

He updated it, you may have a look on it.
I dont currently have time

**Comment by danielwerg** [comment]
> https://github.com/hmes98318/r6s-stats-api
> 
> He updated it, you may have a look on it.
> I dont currently have time

Last I checked that package was simply scraping r6 tracker site, which @BadCoder1337 seem to confirm in other issue.

**Comment by danielwerg** [comment]
Got it working on my private branch, thanks to https://github.com/Seems2Legit/Rainbow-Six-Siege-Player-Stats-API/issues/91#issuecomment-1362191970

![](https://user-images.githubusercontent.com/35052399/209077172-2f6e35f6-f274-4041-9185-c17781fb22eb.png)

Still need to re-factor a lot of code now. Once it's done I will update `master` branch, pr are always welcome :)

Also the comment doesn't _directly_ mention that you need to pass following headers:

```ts
Ubi-AppId: "e3d5ea9e-50bd-43b7-88bf-39794f4e3d40",
Authorization: "ubi_v1 t=TICKET_HERE",
Ubi-SessionId: "UUID_HERE"
```

**Comment by danielwerg** [comment]
https://github.com/danielwerg/r6api.js/commit/76ce63132078fd140e6c26565349a51c9b899bea

Mostly done, not sure if `getProfileSeasonalv2` would be better name, thoughts?
Here is new method names on my v5 branch:

![](https://user-images.githubusercontent.com/35052399/209651848-d9f8fd9f-1563-472d-a9d5-be823150f486.png)

**Comment by RealRunSpace** [comment]
Still doesn't work, tested with both AppsID but still nothing, getting error 500 from server

**Comment by RealRunSpace** [comment]
Yes, that's actually the next branch, the old one didn't requiere the App ID for anything (I think)

**Comment by danielwerg** [comment]
> Yes, that's actually the next branch, the old one didn't requiere the App ID for anything (I think)

Well I will need a snippet of code to reproduced that because it's work fine for me.

**Comment by FelixWehrfeld** [comment]
@RealRunSpace @danielwerg are there any update?

**Comment by RealRunSpace** [comment]
It does work perfectly since lot of time ago

**Comment by FelixWehrfeld** [comment]
@RealRunSpace ohh, then I'm making something wrong... Maybe u can come to the Discord general chat real quick so i can clear up some things?

**Comment by RealRunSpace** [comment]
I'm not at home currently sorry, but someone will help you soon don't worry

**Comment by FelixWehrfeld** [comment]
I think i figured out the problem. There was no npm update ;)

**Comment by FelixWehrfeld** [comment]
Edit: I'm now facing a weird "Internal Server Error". More on Discord if somebody wants to help ty :)
